finetune/target_models/classifier.py

```diff
@@ -209,3 +234,156 @@ class NoisyClassifier(Classifier, BaseModel):
+    :param \**kwargs: key-value pairs of config items to override.
+    """
+
+    def __init__(self, **kwargs):
+        # This is somewhat messy because we also get the default config in the base model init,
+        # but needed to set max_length properly based on max_chunk_length and max_num_chunks
+        default_conf = get_default_config()
+
+        # For now, the only supported base model is LongDocBERT
+        kwargs["base_model"] = LongDocBERT
+
+        # Does not make sense to chunk long sequences
+        kwargs["chunk_long_sequences"] = False
+
+        # Modify max_length based on max_num_chunks
+        # Doing this before initializing anything to avoid shape mismatches with base model
+        # Forcibly setting max_chunk_length when set to auto is a hack
+        max_length = kwargs.get("max_length", default_conf["max_length"])
+        if max_length == "auto":
+            max_length = 128
+        kwargs["max_chunk_length"] = max_length
+        kwargs["max_length"] = max_length * kwargs.get("max_num_chunks", default_conf["max_num_chunks"])
+
+        super().__init__(**kwargs)
+        print(self.config)
+
+        # Make sure that the aggregation/pooling function across chunks is supported
+        # Concat is concatenation of mean and max
+        allowed_pool_fns = {"mean", "max", "attention", "lstm", "concat"}
+        if self.config.chunk_pool_fn not in allowed_pool_fns:
+            raise ValueError(
+                f"chunk_pool_fn={self.config.chunk_pool_fn} "
+                f"not in allowed_fns={allowed_pool_fns}"
+            )
+
+    def featurize(self, X, **kwargs):
+        """
+        Embeds inputs in learned feature space. Can be called before or after calling :meth:`finetune`.
+        :param X: list or array of text to embed.
+        :returns: np.array of features of shape (n_examples, embedding_size).
+        """
+        return super().featurize(X, **kwargs)
+
+    def predict(self, X, context=None, **kwargs):
+        """
+        Produces a list of most likely class labels as determined by the fine-tuned model.
+        :param X: list or array of text to embed.
+        :returns: list of class labels.
+
+        Chunk idx for prediction.  Dividers at `step_size` increments.
+        [  1  |  1  |  2  |  3  |  3  ]
+        """
+        return super().predict(X, context=context, **kwargs)
+
+    def _predict(self, zipped_data, probas=False, **kwargs):
+        """
+        For now, directly copied from Classifier model. Not sure how this logic
+        to process long sequences works with the new long doc classifier
+        """
+        all_labels = []
+        all_probs = []
+        doc_probs = []
+        for (
+            _,
+            _,
+            start_of_doc,
+            end_of_doc,
+            _,
+            proba,
+            _,
+            _,
+        ) in self.process_long_sequence(zipped_data, **kwargs):
```

Comment:
You're going to want to nuke this loop with process_long_sequence()

Reply:
Yea for now I had the implementation identical to the normal `Classifier` model, but I wasn't sure if this was needed or not